### PR TITLE
Migrate server/test_models.py to PyTest

### LINKED ISSUE #3592: Convert bodhi/tests/server/test_models.py to PyTest
[bodhi/tests/server/test_models.py](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/tests/server/test_models.py) still contains classes that derive from `unittest.TestCase`. We use PyTest, so they could be migrated to not derive from `unittest.TestCase`, or derive from `bodhi.tests.server.base.BasePyTestCase` in case they need to use the test database or WSGI app. 

This also allows the use of the simpler assert constructs instead of the self.assert*() methods. Instead of unittest `setUp()` or `tearDown()` methods, PyTest uses `setup_method(...)` or `teardown_method(...)`, or analog methods to setup/teardown stuff for different scopes.

The following PR is an example of the changes that might need to be done to this file:
#3612

### PR DESCRIPTION
**Pull request by ryanlerch**
Fixes #3592

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

### CODE REVIEW

File: bodhi/tests/server/test_models.py [hunk 1]
```diff
@@ -712,90 +706,86 @@ class TestRelease(ModelTest):
...
         self.obj.name = 'f-11'
 
-        self.assertEqual(self.obj.setting_prefix, 'f11')
+        assert self.obj.setting_prefix == 'f11'
 
     @mock.patch.dict(config, {'f11.status': "It's doing just fine, thanks for asking"})
     def test_setting_status_found(self):
         """Assert correct return value from the setting_status property when config is found."""
-        self.assertEqual(self.obj.setting_status, "It's doing just fine, thanks for asking")
+        assert self.obj.setting_status, "It's doing just fine == thanks for asking"
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.setting_status == "It's doing just fine, thanks for asking"
```

File: bodhi/tests/server/test_models.py [hunk 2]
```diff
@@ -1852,9 +1827,7 @@ def test_upgrade_in_stable(self):
         update.release.package_manager = PackageManager.dnf
         update.release.testing_repository = 'updates-testing'
 
-        self.assertEqual(
-            update.install_command,
-            'sudo dnf upgrade --advisory={}'.format(update.alias))
+        assert update.install_command == 'sudo dnf upgrade --advisory={}'.format(update.alias)
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]: Use F-string instead of `format`

File: bodhi/tests/server/test_models.py [hunk 3]
```diff
@@ -1876,16 +1848,14 @@ def test_newpackage_in_stable(self):
         update.release.package_manager = PackageManager.dnf
         update.release.testing_repository = 'updates-testing'
 
-        self.assertEqual(
-            update.install_command,
-            r'sudo dnf install --advisory={} \*'.format(update.alias))
+        assert update.install_command == r'sudo dnf install --advisory={} \*'.format(update.alias)
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py:1851]: Use F-string instead of `format`

File: bodhi/tests/server/test_models.py [hunk 4]
```diff
@@ -2189,7 +2156,7 @@ def test_autokarma_update_reaching_stable_karma(self):
         update.comment(self.db, 'testing', author='hunter2', karma=1)
 
         # meets_testing_requirement() should return True since the karma threshold has been reached
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 5]
```diff
@@ -2204,7 +2171,7 @@ def test_critpath_14_days_negative_karma(self):
         # critpath.
         update.comment(self.db, 'testing', author='bro', karma=1)
 
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 6]
```diff
@@ -2215,7 +2182,7 @@ def test_critpath_14_days_no_negative_karma(self):
         update.date_testing = datetime.utcnow() - timedelta(days=15)
         update.stable_karma = 1
 
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 7]
```diff
@@ -2230,7 +2197,7 @@ def test_critpath_karma_2_met(self):
             # karma. So let's get it a second one.
             update.comment(self.db, 'testing', author='ham', karma=1)
 
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 8]
```diff
@@ -2241,7 +2208,7 @@ def test_critpath_karma_2_required(self):
         # Despite meeting the stable_karma, the function should still not mark this as meeting
         # testing requirements because critpath packages have a higher requirement for minimum
         # karma.
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 9]
```diff
@@ -2251,7 +2218,7 @@ def test_critpath_negative_karma(self):
         update = model.Update.query.first()
         update.critpath = True
         update.comment(self.db, 'testing', author='enemy', karma=-1)
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 10]
```diff
@@ -2264,7 +2231,7 @@ def test_karma_2_met(self):
         # karma. So let's get it a second one.
         update.comment(self.db, 'testing', author='ham', karma=1)
 
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 11]
```diff
@@ -2276,7 +2243,7 @@ def test_non_autokarma_update_below_stable_karma(self):
 
         # meets_testing_requirement() should return False since the karma threshold has not been
         # reached (note that this Update does not have any karma).
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 12]
```diff
@@ -2289,7 +2256,7 @@ def test_non_autokarma_update_reaching_stable_karma(self):
         update.stable_karma = 1
 
         # meets_testing_requirement() should return True since the karma threshold has been reached
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 13]
```diff
@@ -2304,7 +2271,7 @@ def test_test_gating_faild_no_testing_requirements(self):
         update.comment(self.db, 'I found $100 after applying this update.', karma=1,
                        author='bowlofeggs')
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 14]
```diff
@@ -2319,7 +2286,7 @@ def test_test_gating_queued_no_testing_requirements(self):
...
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 15]
```diff
@@ -2334,7 +2301,7 @@ def test_test_gating_running_no_testing_requirements(self):
...
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 16]
```diff
@@ -2349,7 +2316,7 @@ def test_test_gating_missing_testing_requirements(self):
         update.comment(self.db, 'I found $100 after applying this update.', karma=1,
                        author='bowlofeggs')
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 17]
```diff
@@ -2364,7 +2331,7 @@ def test_test_gating_waiting_testing_requirements(self):
         update.comment(self.db, 'I found $100 after applying this update.', karma=1,
                        author='bowlofeggs')
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 18]
```diff
@@ -2379,7 +2346,7 @@ def test_test_gating_off(self):
         update.comment(self.db, 'I found $100 after applying this update.', karma=1,
                        author='bowlofeggs')
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 19]
```diff
@@ -2389,7 +2356,7 @@ def test_time_in_testing_met(self):
         update.date_testing = datetime.utcnow() - timedelta(days=8)
         update.stable_karma = 10
 
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 20]
```diff
@@ -2399,7 +2366,7 @@ def test_time_in_testing_unmet(self):
         update.date_testing = datetime.utcnow() - timedelta(days=6)
         update.stable_karma = 10
 
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 21]
```diff
@@ -2460,13 +2427,13 @@ def test_autokarma_not_nullable(self):
         For history about why this is important, see
         https://github.com/fedora-infra/bodhi/issues/1048
         """
-        self.assertEqual(model.Update.__table__.columns['autokarma'].nullable, False)
+        assert model.Update.__table__.columns['autokarma'].nullable == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not model.Update.__table__.columns['autokarma'].nullable
```

File: bodhi/tests/server/test_models.py [hunk 22]
```diff
@@ -2669,9 +2632,9 @@ def test_days_to_stable_meets_testing_requirements(self):
         update.comment(self.db, 'I found $100 after applying this update.', karma=1,
                        author='bowlofeggs')
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, True)
+        assert update.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 23]
```diff
@@ -2680,10 +2643,10 @@ def test_days_to_stable_not_meets_testing_requirements_no_date_testing(self):
         """
         update = self.get_update()
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, False)
-        self.assertEqual(update.date_testing, None)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 24]
```diff
@@ -2694,9 +2657,9 @@ def test_days_to_stable_not_meets_testing_requirements_with_date_testing(self):
         update = self.get_update()
         update.date_testing = datetime.utcnow() + timedelta(days=-4)
         # Assert that our preconditions from the docblock are correct.
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 25]
```diff
@@ -2712,10 +2675,10 @@ def test_days_to_stable_zero(self):
         update.test_gating_status = TestGatingStatus.failed
 
         update.date_testing = datetime.utcnow() + timedelta(days=-8)
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 26]
```diff
@@ -2731,40 +2694,38 @@ def test_days_to_stable_positive(self):
         update.test_gating_status = TestGatingStatus.failed
 
         update.date_testing = datetime.utcnow() + timedelta(days=-3)
-        self.assertEqual(update.meets_testing_requirements, False)
+        assert update.meets_testing_requirements == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not update.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 27]
```diff
@@ -2731,40 +2694,38 @@ def test_days_to_stable_positive(self):
...
-        self.assertEqual(self.obj.requested_tag, self.obj.release.candidate_tag)
+        assert self.obj.requested_tag == self.obj.release.candidate_tag
 
     def test_side_tag_locked_false(self):
         """Test the side_tag_locked property when it is false."""
         self.obj.status = model.UpdateStatus.side_tag_active
         self.obj.request = None
 
-        self.assertIs(self.obj.side_tag_locked, False)
+        assert self.obj.side_tag_locked is False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not self.obj.side_tag_locked
```

File: bodhi/tests/server/test_models.py [hunk 28]
```diff
@@ -2731,40 +2694,38 @@ def test_days_to_stable_positive(self):
...
 
     def test_side_tag_locked_true(self):
         """Test the side_tag_locked property when it is true."""
         self.obj.status = model.UpdateStatus.side_tag_active
         self.obj.request = model.UpdateRequest.stable
 
-        self.assertIs(self.obj.side_tag_locked, True)
+        assert self.obj.side_tag_locked is True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.side_tag_locked
```

File: bodhi/tests/server/test_models.py [hunk 29]
```diff
@@ -2782,16 +2743,15 @@ def test_modify_bugs_stable_close(self, comment, close):
...
         # Make sure close() was called correctly.
-        self.assertEqual([c[1][0] for c in close.mock_calls], [1, 2])
-        self.assertEqual(all(
-            ['to the Fedora 11 stable repository' in c[2]['comment'] for c in close.mock_calls]),
-            True)
-        self.assertEqual(all(
+        assert [c[1][0] for c in close.mock_calls] == [1, 2]
+        assert all(
+            ['to the Fedora 11 stable repository' in c[2]['comment']
+                for c in close.mock_calls]) == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py:2751]:
```suggestion
                for c in close.mock_calls])
```

File: bodhi/tests/server/test_models.py [hunk 30]
```diff
@@ -2782,16 +2743,15 @@ def test_modify_bugs_stable_close(self, comment, close):
...
+        assert all(
             [c[2]['versions']['TurboGears'] == 'TurboGears-1.0.8-3.fc11'
-                for c in close.mock_calls]),
-            True)
+                for c in close.mock_calls]) == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py:2754]:
```suggestion
                for c in close.mock_calls])
```

File: bodhi/tests/server/test_models.py [hunk 31]
```diff
@@ -2808,12 +2768,12 @@ def test_modify_bugs_stable_no_close(self, comment, close):
...
 
         # Make sure bugs number 1 and 2 were commented on correctly.
-        self.assertEqual([c[1][0] for c in comment.mock_calls], [1, 2])
-        self.assertEqual(all(
-            ['pushed to the Fedora 11 stable repository' in c[1][1] for c in comment.mock_calls]),
-            True)
+        assert [c[1][0] for c in comment.mock_calls] == [1, 2]
+        assert all(
+            ['pushed to the Fedora 11 stable repository' in c[1][1]
+                for c in comment.mock_calls]) == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py:2774]:
```suggestion
                for c in comment.mock_calls])
```

File: bodhi/tests/server/test_models.py [hunk 32]
```diff
@@ -3432,8 +3384,8 @@ def test_set_request_stable_after_week_in_testing(self):
         self.obj.comment(
             self.db, 'This update has been pushed to testing.', author='bodhi')
         self.obj.date_testing = self.obj.comments[-1].timestamp - timedelta(days=7)
-        self.assertEqual(self.obj.days_in_testing, 7)
-        self.assertEqual(self.obj.meets_testing_requirements, True)
+        assert self.obj.days_in_testing == 7
+        assert self.obj.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 33]
```diff
@@ -3464,8 +3416,8 @@ def test_set_request_stable_when_release_is_frozen(self):
         self.obj.comment(
             self.db, u'This update has been pushed to testing.', author=u'bodhi')
         self.obj.date_testing = self.obj.comments[-1].timestamp - timedelta(days=7)
-        self.assertEqual(self.obj.days_in_testing, 7)
-        self.assertEqual(self.obj.meets_testing_requirements, True)
+        assert self.obj.days_in_testing == 7
+        assert self.obj.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 34]
```diff
@@ -3586,16 +3538,16 @@ def test_has_stable_comment_at_7_days_after_bodhi_comment(self):
         self.obj.comment(
             self.db, 'This update has been pushed to testing.', author='bodhi')
         self.obj.date_testing = self.obj.comments[-1].timestamp - timedelta(days=7)
-        self.assertEqual(self.obj.days_in_testing, 7)
+        assert self.obj.days_in_testing == 7
         # The update should be eligible to receive the testing_approval_msg now.
-        self.assertEqual(self.obj.meets_testing_requirements, True)
+        assert self.obj.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 35]
```diff
@@ -3586,16 +3538,16 @@ def test_has_stable_comment_at_7_days_after_bodhi_comment(self):
...
         # Add the testing_approval_message
         text = str(config.get('testing_approval_msg'))
         self.obj.comment(self.db, text, author='bodhi')
 
         # met_testing_requirement() should return True since Bodhi has commented on the Update to
         # say that it can now be pushed to stable.
-        self.assertEqual(self.obj.has_stable_comment, True)
+        assert self.obj.has_stable_comment == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.has_stable_comment
```

File: bodhi/tests/server/test_models.py [hunk 36]
```diff
@@ -3607,12 +3559,12 @@ def test_has_stable_comment_at_7_days_before_bodhi_comment(self):
         self.obj.comment(
             self.db, 'This update has been pushed to testing.', author='bodhi')
         self.obj.date_testing = self.obj.comments[-1].timestamp - timedelta(days=7)
-        self.assertEqual(self.obj.days_in_testing, 7)
+        assert self.obj.days_in_testing == 7
         # The update should be eligible to receive the testing_approval_msg now.
-        self.assertEqual(self.obj.meets_testing_requirements, True)
+        assert self.obj.meets_testing_requirements == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.meets_testing_requirements
```

File: bodhi/tests/server/test_models.py [hunk 37]
```diff
@@ -3607,12 +3559,12 @@ def test_has_stable_comment_at_7_days_before_bodhi_comment(self):
...
 
         # Since bodhi hasn't added the testing_approval_message yet, this should be False.
-        self.assertEqual(self.obj.has_stable_comment, False)
+        assert self.obj.has_stable_comment == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not self.obj.has_stable_comment
```

File: bodhi/tests/server/test_models.py [hunk 38]
```diff
@@ -3637,7 +3589,7 @@ def test_has_stable_comment_with_karma_after_bodhi_comment(self):
 
         # met_testing_requirement() should return True since Bodhi has commented on the Update to
         # say that it can now be pushed to stable.
-        self.assertEqual(self.obj.has_stable_comment, True)
+        assert self.obj.has_stable_comment == True
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert self.obj.has_stable_comment
```

File: bodhi/tests/server/test_models.py [hunk 39]
```diff
@@ -3651,49 +3603,48 @@ def test_has_stable_comment_with_karma_before_bodhi_comment(self):
...
+        assert self.obj.days_in_testing == 1
         # Now let's add some karma to get it to the required threshold
         self.obj.comment(self.db, 'testing', author='hunter1', karma=1)
         self.obj.comment(self.db, 'testing', author='hunter2', karma=1)
         self.obj.comment(self.db, 'testing', author='hunter3', karma=1)
 
         # met_testing_requirement() should return False since Bodhi has not yet commented on the
         # Update to say that it can now be pushed to stable.
-        self.assertEqual(self.obj.has_stable_comment, False)
+        assert self.obj.has_stable_comment == False
```
Comment (Zlopez) [inline bodhi/tests/server/test_models.py]:
```suggestion
        assert not self.obj.has_stable_comment 
```

**Review by Zlopez — COMMENTED** [review]
Few suggestions

**Review by Zlopez — COMMENTED** [review]
Few suggestions

File: bodhi/tests/server/test_models.py (hunk 2 above)
Comment (ryanlerch) [reply bodhi/tests/server/test_models.py]: fixed. thanks.

**Review by Zlopez — APPROVED** [review]

**Pull request merged**